### PR TITLE
feat(perf): Add N+1 API Calls performance issue detector

### PR DESCRIPTION
**Pull request by gggritso**
Adds a very rudimentary (for now) detector for N+1 API Calls. The idea is to detect a large number (5) concurrent (starting within 5ms of each other) HTTP network calls. If this sounds like it's _just_ detecting concurrent API calls, rather than N+1, you are correct. I think we'll want to improve this by setting some kind of similarity criteria for the network calls (i.e., similar-looking URLs) but this should be enough to look and learn for now.

Also is pretty generic, we can extend this to other `op`s if we so choose.

Closes PERF-1695.

### CODE REVIEW

**Review by k-fish — APPROVED** [review]
Looks good, let's ship it 🚢 . Now that I have had a chance to look at the code I'm not sure the threshold trick for similar start times will be super reliable, I've considered it before, and it seems to be pretty dependent on the system. We'll maybe have to do more tuning on this one in particular by observing the detected outcomes. 

We can do that next after we try this out first, though. Nicely done George 👏

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -85,6 +86,13 @@ def get_default_detection_settings():
                 "allowed_span_ops": ["ui.long-task", "ui.sentry.long-task"],
             }
         ],
+        DetectorType.N_PLUS_ONE_SPANS: [
+            {
+                "count": 5,
+                "start_time_threshold": 5.0,  # ms
+                "allowed_span_ops": ["http.client"],
```
[the remark this answers is not in the document]
Comment (gggritso) [reply src/sentry/utils/performance_issues/performance_detection.py]: I was focusing on the front-end use case, but I added `"http.server"` too, and see what it reports, if anything!

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 2]
```diff
@@ -411,6 +420,64 @@ def visit_span(self, span: Span):
...
+            return
+
+        op, span_id, op_prefix, span_duration, settings = settings_for_span
+
+        start_time_threshold = timedelta(milliseconds=settings.get("start_time_threshold", 0))
+        count = settings.get("count", 10)
+
+        fingerprint = fingerprint_span_op(span)
+        if not fingerprint:
+            return
```
[the remark this answers is not in the document]
Comment (gggritso) [reply src/sentry/utils/performance_issues/performance_detection.py:525]: Can-do!

**Comment by gggritso** [comment]
@k-fish makes sense, thanks! I'll definitely babysit this one, so I'll be tweaking it over this coming week. I'm also wondering if maybe 5 is too low a required count, for example

**Comment by k-fish** [comment]
@gggritso it's fine as long as it doesn't break ingest, counting more or less just changes a number we send in metrics, so start with a best guess and tune from there.

**Pull request merged**